.github/workflows/test.yml

```diff
@@ -2,64 +2,37 @@ name: 'Test'
 
 on: [push, pull_request, workflow_dispatch]
 
+
 jobs:
   test:
-    runs-on: ubuntu-18.04
+    runs-on: ${{ matrix.os }}
     timeout-minutes: 10
+    strategy:
+      matrix:
+        os: [ubuntu-latest, windows-latest, macos-latest]
+        python-version: ['3.8', '3.9', '3.10']
 
     steps:
-      - name: Check out code
-        uses: actions/checkout@v2
-
-      - name: Setup Python
-        uses: actions/setup-python@v2
-        with:
-          python-version: "3.9"
-
-      - name: Cache pip
-        uses: actions/cache@v2
-        with:
-          path: ~/.cache/pip
-          key: ${{ runner.os }}-pip-${{ hashFiles('**/requirements.txt') }}
-          restore-keys: |
-            ${{ runner.os }}-pip-
 
-      - name: Install dependencies
-        run:
-          python -m pip install -r tests/requirements.txt
+    - uses: actions/checkout@v2
 
-      - name: Run pyright tests
-        uses: gramster/pyright-action@main
-        with:
-          project: pyrighttestconfig.json
-          warn-partial: true
+    - name: Set up Python ${{ matrix.python-version }}
+      uses: actions/setup-python@v2
+      with:
+        python-version: ${{ matrix.python-version }}
 
-      - name: Run pytest
-        run:
-          pytest tests/pandas
-          
-      - name: Run mypy
-        run:
-          mypy tests/pandas typings/pandas
-          
-      - name: Build wheel and install and remove typings
-        run: |
-          python setup.py build bdist_wheel
-          find ./dist/*.whl | xargs pip install
-          rm -rf typings
-          pip install pyright
-      
-      # - name: run pyright command line
-      #   run:
-      #     pyright -p pyrightdistconfig.json --dependencies
+    - name: Upgrade pip
+      run: |
+        python -m pip install --upgrade pip
+    - name: Install Poetry and project dependencies
+      run: |
+        pip install poetry
+        poetry install
+    - name: Run Pyright
+      run: poetry run pyright
 
-      # - name: Run pyright against dist
-      #   uses: gramster/pyright-action@main
-      #   with:
-      #     project: pyrightdistconfig.json
-      #     extra-args: --dependencies
-      #     warn-partial: true
+    - name: Run Pytest
+      run: poetry run pytest
 
-      - name: Run mypy against dist
-        run:
-          mypy tests/pandas
+    - name: Run MyPy
+      run: poetry run mypy pandas-stubs tests
```

Comment:
One issue that I have run into with respect to using the more traditional method of a build using `setup.py` is that the wheel that is built, when installed, isn't passing the tests.  So if you look at https://github.com/pandas-dev/pandas-stubs/blob/main/.github/workflows/test.yml , you will see this in there:
```
     - name: Build wheel and install and remove typings
        run: |
          python setup.py build bdist_wheel
          find ./dist/*.whl | xargs pip install
          rm -rf typings
          pip install pyright
      
      # - name: run pyright command line
      #   run:
      #     pyright -p pyrightdistconfig.json --dependencies

      # - name: Run pyright against dist
      #   uses: gramster/pyright-action@main
      #   with:
      #     project: pyrightdistconfig.json
      #     extra-args: --dependencies
      #     warn-partial: true

      - name: Run mypy against dist
        run:
          mypy tests/pandas
```

The problem right now is that `pyright` is failing with this, and there is a long discussion at https://github.com/microsoft/pyright/issues/3540 , there seems to be an issue with the presence of the `py.typed` file that is interpreted differently on MacOS versus Windows and Linux.

In any case, adding the tests here to build the wheel, install it, and then test pyright and mypy against the installed wheel is something that should be done.

Reply:
I had a lot of trouble figuring out the "py.typed" with "partial\n" string it was causing the problem, at least in my environment. not just in wheel installed case. So for a while, I just delete the file to show the poetry idea. I have the three Local OS here, I'm gonna test and see if I can help to fix the bug in the official pyright repository. To add the test with the wheel is not gonna be hard when the pyright is gonna work properly.

Reply:
So it turns out that if I remove `py.typed`, then everything works locally and with the installed wheel.  That version is now in `main`.  So remove `py.typed` from your version and it should be OK